### ISSUE TITLE
[no-for-loop] incorrect fix for string

### ISSUE THREAD

**Issue by JounQin** [sindresorhus/eslint-plugin-unicorn#1142]
```js
for (let i = 0; i < needle.length; i++) {
  if (haystack[i] !== needle[i]) {
    return false
  }
}
```

```js
// fixed
for (const [i, element] of needle.entries()) {
  if (haystack[i] !== element) {
    return false
  }
}
```

```js
// correct fix
for (const [i, element] of [...needle].entries()) {
  if (haystack[i] !== element) {
    return false
  }
}
```

**Comment by JounQin** [comment]
duplicate of #738
